### ISSUE TITLE
Docker container failing to initialise.

### ISSUE THREAD

**Issue by gitgoggles** [silverbulletmd/silverbullet#331]
I've attempted to setup the docker container, fresh, on an Unraid server using Docker Compose. The container is struggling to start and I'm seeing this in the container logs:

### docker logs silverbullet
~~~
Going to start SilverBullet binding to 0.0.0.0:3000
Serving pages from /space
Error initializing database Error: Permission denied (os error 13), open '/space/data.db'
    at Worker.<anonymous> (file:///silverbullet.js:186745:28)
    at innerInvokeEventListeners (deno:ext/web/02_event.js:755:9)
    at invokeEventListeners (deno:ext/web/02_event.js:795:7)
    at dispatch (deno:ext/web/02_event.js:664:11)
    at Worker.dispatchEvent (deno:ext/web/02_event.js:1043:14)
    at Worker.#pollMessages (deno:runtime/js/11_workers.js:197:14)
HTTP Server error Error: Not initialized
    at Worker.<anonymous> (file:///silverbullet.js:186745:28)
    at innerInvokeEventListeners (deno:ext/web/02_event.js:755:9)
    at invokeEventListeners (deno:ext/web/02_event.js:795:7)
    at dispatch (deno:ext/web/02_event.js:664:11)
    at Worker.dispatchEvent (deno:ext/web/02_event.js:1043:14)
    at Worker.#pollMessages (deno:runtime/js/11_workers.js:197:14)
~~~

### compose.yml
~~~
services:
    silverbullet:
        image: zefhemel/silverbullet
        container_name: silverbullet
        restart: unless-stopped
        volumes:
          - '/mnt/user/appdata/silverbullet:/space'
        ports:
        - 3002:3000
~~~

### permissions
If I ''ls -la'' on my appdata directory, I can see docker has created the silverbullet directory:
~~~
drwxr-xr-x 1 root   root     6 Jan 22 15:37 silverbullet/
~~~

I'm fairly new to github, docker and linux in general so let me know if there is any more information I can provide. I can see suggestion of a permissions issue in the logs, but I'm not super sure on how to resolve it.

**Comment by zefhemel** [comment]
Does whatever user this container runs as have write permission to that silverbullet directory? It doesn't seem that way.

**Comment by gitgoggles** [comment]
Thank you, you were correct. After some experimentation, I got it to run via one of the following (sorry I can't be more specific).

1. Running `chown nobody:users silverbullet/`

2. Adding PUID, PGID and UMASK environment variables to the docker compose config:
~~~
services:
    silverbullet:
        image: zefhemel/silverbullet
        container_name: silverbullet
        restart: unless-stopped
        environment:
          - PUID=99
          - PGID=100
          - UMASK=022
        volumes:
          - '/mnt/user/appdata/silverbullet:/space'
        ports:
        - 3002:3000
~~~
3. Running `chmod +w silverbullet/`

This was the order in which I made changes, recreating the container at each step. It didn't start working until the last one, so perhaps the last step is all that's needed, but I can't be certain. Thank you for your suggestion.
